tests/test_index.py

```diff
@@ -149,6 +149,38 @@ def test_similarity_errors(self, indexed_img_tbl: pxt.Table, small_img_tbl: pxt.
             _ = t.order_by(t.split.similarity(sample_img)).limit(1).collect()
         assert "was created without the 'image_embed' parameter" in str(exc_info.value).lower()
 
+    def test_add_index_after_drop(self, small_img_tbl: pxt.Table, reload_tester: ReloadTester) -> None:
+        """ Test the an index with the same name can be added after the previous one is dropped """
+        skip_test_if_not_installed('transformers')
+        t = small_img_tbl
+        sample_img = t.select(t.img).head(1)[0, 'img']
+        t.add_embedding_index('img', idx_name='clip_idx', image_embed=clip_img_embed, string_embed=clip_text_embed)
+        orig_res = t.select(t.img.localpath).order_by(t.img.similarity(sample_img, idx='clip_idx'), asc=False).limit(3).collect()
+        t.revert()
+        # creating an index with the same name again after a revert should be successful
+        t.add_embedding_index('img', idx_name='clip_idx', image_embed=clip_img_embed, string_embed=clip_text_embed)
+        res = t.select(t.img.localpath).order_by(t.img.similarity(sample_img, idx='clip_idx'), asc=False).limit(3).collect()
+        assert_resultset_eq(orig_res, res, True)
+        t.revert()
+        # should be true even after reloading from persistence
+        reload_tester.run_reload_test()
+        t = pxt.get_table('small_img_tbl')
+        t.add_embedding_index('img', idx_name='clip_idx', image_embed=clip_img_embed, string_embed=clip_text_embed)
+        res = t.select(t.img.localpath).order_by(t.img.similarity(sample_img, idx='clip_idx'), asc=False).limit(3).collect()
+        assert_resultset_eq(orig_res, res, True)
```

Comment:
This is what the reload tester is for.

Reply:
In in this case, I am trying to compare the result of the similarity query with original index, and then the new index recreated after revert/drop, without any reload in between. In this testcase, at the time of reload, there's no index. The purpose of reload is to make sure drop_index cleans up the state in persistence and allows creating that index again. Using the index in the query is a sanity check to make sure the creation was correct (which is tested elsewhere, but not in this particular sequence). HTH.
<!-- Reviewable comment -OEBWgf00bPh6txylc82:bj3rpto -->
<!-- Sent from Reviewable.io -->

Reply:
@mkornacker I don't think so? Here we're validating that the in-memory query results are stable over the intervening operations. That's a different scenario from what the reload tester checks for.
<!-- Reviewable comment -OEQYs_O300bn7naoFCw:b1qs7yb -->
<!-- Sent from Reviewable.io -->

Reply:
Yes, true here, my mistake.